File: frontend/src/app/core/services/book/models/book-details.dto.ts
import { EbookFormat } from '../../../../shared/enums/ebook-format';
import { Genre } from '../../../../shared/enums/genre';
import { Language } from '../../../../shared/enums/language';
import { ShelfDto } from '../../../../shared/models/shelf.dto';

export interface BookDetailsDto {
  readonly id: number;
  readonly title: string;
  readonly author: string;
  readonly publisher: string;
  readonly isbn: string;
  readonly publicationYear: number;
  readonly genre: Genre;
  readonly pageCount: number;
  readonly language: Language;
  readonly averageRating: number;
  readonly numberOfRatings: number;
  readonly description: string;
  readonly imagePath: string;
  readonly isEbook: boolean;
  readonly ebookFormat: EbookFormat | null;
  readonly ebookFileSize: number | null;
  readonly ebookLink: string;
  readonly shelves: ShelfDto[];
}

File: backend/src/main/java/com/example/bookhub/dto/BookDetailsDto.java
package com.example.bookhub.dto;

import com.example.bookhub.enums.EbookFormat;
import com.example.bookhub.enums.Genre;
import com.example.bookhub.enums.Language;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Data;

import java.util.List;

@Data
public class BookDetailsDto {

    private Long id;

    private String title;

    private String author;

    private String publisher;

    private String isbn;

    private int publicationYear;

    private Genre genre;

    private int pageCount;

    private Language language;

    private double averageRating;

    private int numberOfRatings;

    private String description;

    private String imagePath;

    @JsonProperty("isEbook")
    private boolean isEbook;

    private EbookFormat ebookFormat;

    private Double ebookFileSize;

    private String ebookLink;

    private List<ShelfDto> shelves;
}
